finagle-core/src/main/scala/com/twitter/finagle/dispatch/ServerDispatcher.scala

```diff
@@ -103,8 +114,30 @@ class SerialServerDispatcher[Req, Rep](
     service.close()
   }
 
-  protected def dispatch(req: Req, eos: Promise[Unit]) =
+  override def dispatch(req: Req, eos: Promise[Unit]) =
     service(req) ensure eos.setDone()
 
-  protected def handle(rep: Rep) = trans.write(rep)
+  override def handle(rep: Rep) = trans.write(rep)
+}
+
+/**
+  * A wrapper type around a 
```

Comment:
"A wrapper type" isn't terribly descriptive.  How about something like:

"Wraps a ServerDispatcher, adding annotations when a request is read from the wire, and later is written to the wire."